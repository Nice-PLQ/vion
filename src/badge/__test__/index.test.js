import { shallowMount } from '@vue/test-utils';
import Badge from '../badge.vue';

describe('Badge', () => {
  const wrapper = shallowMount(Badge, {
    slots: {
      default: '99+'
    },
    props: {
      backgroundColor: '#B0B3BF',
      textColor: '#000',
      size: '16px'
    }
  });

  test('类型', async () => {
    await wrapper.setProps({ type: 'dot' });
    expect(wrapper.classes()).toContain('v-badge_dot');
    expect(wrapper.element).toMatchSnapshot();

    await wrapper.setProps({ type: 'capsule' });
    expect(wrapper.classes()).toContain('v-badge_capsule');
    expect(wrapper.element).toMatchSnapshot();
  });

  test('大小', () => {
    expect(wrapper.element.style.fontSize).toBe('16px');
  });

  test('颜色', () => {
    expect(wrapper.element.style.backgroundColor).toBe('rgb(176, 179, 191)');
    expect(wrapper.element.style.color).toBe('rgb(0, 0, 0)');
  });

  test('文本', () => {
    expect(wrapper.text()).toBe('99+');
  });

  test('click事件触发', async () => {
    await wrapper.trigger('click');
    expect(wrapper.emitted('click')).toBeTruthy();
  });
});
