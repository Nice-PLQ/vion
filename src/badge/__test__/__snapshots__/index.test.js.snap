// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge 类型 1`] = `
<div
  class="v-badge v-badge_dot"
  style="background-color: rgb(176, 179, 191); color: rgb(0, 0, 0); font-size: 16px;"
>
  <span
    class="v-badge__number"
  >
    
    99+
    
  </span>
</div>
`;

exports[`Badge 类型 2`] = `
<div
  class="v-badge v-badge_capsule"
  style="background-color: rgb(176, 179, 191); color: rgb(0, 0, 0); font-size: 16px;"
>
  <span
    class="v-badge__number"
  >
    
    99+
    
  </span>
</div>
`;
